fabtests/examples/example_rdm.c

```diff
@@ -0,0 +1,507 @@
+	 * the hints requirements. Any applicable providers will be returned as
+	 * a list of fi_info structs (&info). Any info can be selected. In this
+	 * test we select the first fi_info struct. Assuming all hints were set
+	 * appropriately, the first fi_info should be most appropriate.
+	 * The flag FI_SOURCE is set for the server to indicate that the
+	 * address/port refer to source information. This is not set for the
+	 * client because the fields refer to the server, not the caller
+	 * (client).
+	 */
+
+	/*
+	 * (fi_getinfo)(uint32_t version, const char *node,
+	 * 		const char *service, uint64_t flags,
+	 * 		const struct fi_info *hints, struct fi_info **info)
+	 */
+	ret = fi_getinfo(FI_VERSION(1,9), dst_addr, port,
+			dst_addr ? 0 : FI_SOURCE, hints, &info);
+	if (ret) {
+		printf("fi_getinfo error (%d)\n", ret);
+		return ret;
+	}
+
+	if (!dst_addr) {
+		/*
+		 * The server converts the returned fi_info->src_addr to a
+		 * string for the client to use in its command line call.
+		 * This address will get passed into the client's fi_getinfo
+		 * call to resolve the server's address.
+		 */
+		sin = info->src_addr;
+		if (!inet_ntop(sin->sin_family, &sin->sin_addr, str_addr,
+			sizeof(str_addr))) {
+			printf("error converting address to string\n");
+			return -1;
+		}
+
+		printf("Server started with addr %s\n", str_addr);
+	}
+
+	/*
+	 * Initialize our fabric. The fabric network represents a collection of
+	 * hardware and software resources that access a single physical or
+	 * virtual network. All network ports on a system that can communicate
+	 * with each other through their attached networks belong to the same
+	 * fabric.
+	 */
+
+	/*
+	 * fi_fabric(struct fi_fabric_attr *attr,
+	 * 	     struct fid_fabric **fabric, void *context);
+	 */
+	ret = fi_fabric(info->fabric_attr, &fabric, NULL);
+	if (ret) {
+		printf("fi_fabric error (%d)\n", ret);
+		return ret;
+	}
+
+	/*
+	 * Initialize our domain (associated with our fabric). A domain defines
+	 * the boundary for associating different resources together.
+	 */
+
+	/*
+	 * fi_domain(struct fid_fabric *fabric, struct fi_info *info,
+	 * 	     struct fid_domain **domain, void *context);
+	 */
+	ret = fi_domain(fabric, info, &domain, NULL);
+	if (ret) {
+		printf("fi_domain error (%d)\n", ret);
+		return ret;
+	}
+
+	/*
+	 * Initialize our endpoint. Endpoints are transport level communication
+	 * portals which are used to initiate and drive communication. There
+	 * are three main types of endpoints:
+	 * FI_EP_MSG - connected, reliable
+	 * FI_EP_RDM - unconnected, reliable
+	 * FI_EP_DGRAM - unconnected, unreliable
+	 * The type of endpoint will be requested in hints/fi_getinfo.
+	 * Different providers support different types of endpoints. TCP
+	 * supports only FI_EP_MSG but when used with RxM, can support
+	 * FI_EP_RDM. In this application, we requested TCP and FI_EP_RDM
+	 * so we will automatically pick up RxM in order to use RDM with TCP
+	 * hardware.
```

Comment:
Stop comment at Different providers support different types of endpoints.  The rest of the comment is no longer valid.